core/extensions/DllMapDemo/NewLib.cpp

```diff
@@ -0,0 +1,12 @@
+#include <stdio.h>
+
+#if defined(__GNUC__)
+#define EXPORT extern "C" __attribute__((visibility("default")))
+#elif defined(_MSC_VER)
+#define EXPORT extern "C" __declspec(dllexport)
+#endif 
+
+extern "C" EXPORT int NativeSum(int a, int b)
+{
+    return a + b;
+}
```

Comment:
I do not think we should be committing the binary to repo. Instead, I would include a simple instructions (command line) how to build the library on Windows and Unix.